### ISSUE TITLE
Trip is blank in eTools

### ISSUE THREAD

**Issue by marindikanovic** [unicef/etools-issues#918]
The following trip isn't showing in eTools: https://etools.unicef.org/t2f/edit-travel/24800

For the user (Ali Chamseddine), he sees the following details, which includes Office, Supervisor and Section.

**Comment by marindikanovic** [comment]
<img width="1311" alt="screen shot 2017-11-10 at 12 58 37 pm" src="https://user-images.githubusercontent.com/29664134/32741060-f5010db4-c872-11e7-9257-997fc973e081.png">

**Comment by marindikanovic** [comment]
here is the error link: https://etools.unicef.org/api/t2f/travels/24800/

**Comment by reinbach** [comment]
TLDR: Unicode issues

We save the filename using the following snippet of code;

```
from __future__ import unicode_literals
...
    return 'travels/{}/{}/{}'.format(country_name, instance.travel.id, filename)
```

So when we pull the data from the database we're in `u"..."` state, and when we have non-ascii chars in the string and attempting to use in with something like `"{}".format(u"...")` we end up with `UnicodeDecodeError`

When attempting to get the file via AzureStorage a call is made to `make_blob_url` in AzureStorage, but an error (see error below) is occuring because in this instance the attachment to the trip has a filename with non-ascii characters (`u'travels/test/125/travels/lebanon/24800/\u0628\u0631\u0646\u0627\u0645\u062c \u062a\u062f\u0631\u064a\u0628 \u0627\u0644\u0645\u062a\u0627\u0628\u0639\u064a\u0646.pdf'`)

The [snippet of code](https://github.com/Azure/azure-storage-python/blob/09e9f186740407672777d6cb6646c33a2273e1a8/azure/storage/blob/blobservice.py#L182) in the version of AzureStorage that we're using (0.20.2)

It looks like [latest versions](https://github.com/Azure/azure-storage-python/blob/master/azure-storage-blob/azure/storage/blob/baseblobservice.py#L245) of AzureStorage still have this issue.

AzureStorage is reached via `django-storages` where we create the url, a possible fix is to encode `.encode("utf8")` the blob name when calling the `make_blog_url` from `django-storages`. 
https://github.com/robertavram/django-storages/blob/master/storages/backends/azure_storage.py#L120

Error is:
```
[lebanon:lebanon.etools.unicef.org] ERROR   2017-12-07 17:21:29,405 Internal Server Error: /api/t2f/travels/24800/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/viewsets.py", line 83, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 477, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 437, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 474, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/mixins.py", line 58, in retrieve
    return Response(serializer.data)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 527, in data
    ret = super(Serializer, self).data
  File "/usr/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/code/EquiTrack/t2f/serializers/travel.py", line 348, in to_representation
    data = super(TravelDetailsSerializer, self).to_representation(instance)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 496, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 643, in to_representation
    self.child.to_representation(item) for item in iterable
  File "/usr/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 483, in to_representation
    attribute = field.get_attribute(instance)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/fields.py", line 443, in get_attribute
    return get_attribute(instance, self.source_attrs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/fields.py", line 102, in get_attribute
    instance = getattr(instance, attr)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/fields/files.py", line 69, in _get_url
    return self.storage.url(self.name)
  File "/usr/local/lib/python2.7/site-packages/storages/backends/azure_storage.py", line 122, in url
    sas_token=sas_token
  File "/usr/local/lib/python2.7/site-packages/azure/storage/blob/blobservice.py", line 187, in make_blob_url
    blob_name,
UnicodeEncodeError: 'ascii' codec can't encode characters in position 22-27: ordinal not in range(128)
```
